samples/third-party/gltf-loader.js

```diff
@@ -0,0 +1,354 @@
+            for (var semantic in attributes) {
+                var accessorName = attributes[semantic];
+                attribute = accessors[accessorName];
+
+                if (semantic.substring(0, 8) === 'POSITION') {
+                    stride = scene.positionByteStride / AttributeSize[attribute.componentType];
+                    offset = scene.positionByteOffset / AttributeSize[attribute.componentType];
+                    count = attribute.count;
+
+                    for (var i = 0; i < count; ++i) {
+                        // @todo: add vec2 and other(needed?) support
+                        if(scene.positionNumberOfComponents === 3) {
+                            vec4.set(tmpVec4, data[stride * i + offset]
+                                            , data[stride * i + offset + 1]
+                                            , data[stride * i + offset + 2]
+                                            , 1);
+                            vec4.transformMat4(tmpVec4, tmpVec4, matrix);
+                            vec4.scale(tmpVec4, tmpVec4, tmpVec4[3]);
+                            data[stride * i + offset] = tmpVec4[0];
+                            data[stride * i + offset + 1] = tmpVec4[1];
+                            data[stride * i + offset + 2] = tmpVec4[2];
+                        }
+                    }
+                } else if (semantic.substring(0, 6) === 'NORMAL') {
+                    stride = scene.normalByteStride / AttributeSize[attribute.componentType];
+                    offset = scene.normalByteOffset / AttributeSize[attribute.componentType];
+                    count = attribute.count;
+                    mat4.invert(inverseTransposeMatrix, matrix);
+                    mat4.transpose(inverseTransposeMatrix, inverseTransposeMatrix);                    
+
+                    for (var i = 0; i < count; ++i) {
+                        // @todo: add vec2 and other(needed?) support
+                        vec4.set(tmpVec4, data[stride * i + offset]
+                                        , data[stride * i + offset + 1]
+                                        , data[stride * i + offset + 2]
+                                        , 0);
+                        vec4.transformMat4(tmpVec4, tmpVec4, inverseTransposeMatrix);
+                        vec4.normalize(tmpVec4, tmpVec4);
+                        data[stride * i + offset] = tmpVec4[0];
+                        data[stride * i + offset + 1] = tmpVec4[1];
+                        data[stride * i + offset + 2] = tmpVec4[2];
+                    }
+                }
+            }
+            
+            if (data) {
+                scene.vertices = data;
+
+                // Completed loading
+                callback(scene);
+            }
+        });
+    };
+
+    window.loadGltfGeometry = function(url, onload) {
+        
+        // @todo: currently using a relative path to load gl-matrix-min
+        (function(src, callback){
```

Comment:
Do not do this.  For now, just require to user to include gl-matrix-min in a separate script tag above this.

Longer term, I'll look at a client-side build.